src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/Extensions/StringExtensions.cs

```diff
@@ -105,6 +105,30 @@ public static string RemoveBlankLines(this string value)
                 .Trim();
         }
 
+        /// <summary>
+        /// Removes the duplicate string from parameter name to work around the rosyln issue.
+        /// https://github.com/dotnet/roslyn/issues/26292.
+        /// e.g if the value is service-$Catalog-Id-1-$Catalog-Id-1, this will return
+        /// service-$Catalog-Id
+        /// </summary>
+        /// <param name="value">The param name to update.</param>
+        /// <returns>The updated value.</returns>
+        public static string RemoveDuplicateString(this string value)
```

Comment:
nit, more specific naming. RemoveRoslynDuplicateDashedString?

Reply:
its not just dashed but any special characters that can't be  used in a param name so i have for now just picked few which could be used. but i can name it to RemoveRoslynDuplicateString

---
In reply to: [253635331](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/198#discussion_r253635331) [](ancestors = 253635331)